javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -211,16 +214,20 @@ class LinkedList[E]() extends AbstractSequentialList[E]
   }
 
   def peekFirst(): E =
-    head.value
+    if (head eq null) null.asInstanceOf[E]
+    else head.value
 
   def peekLast(): E =
-    peekFirst()
+    if (last eq null) null.asInstanceOf[E]
+    else last.value
 
   def pollFirst(): E =
-    removeFirst()
+    if (isEmpty()) null.asInstanceOf[E]
+    else removeFirst()
 
   def pollLast(): E =
-    removeLast()
+    if (isEmpty) null.asInstanceOf[E]
+    else removeLast()
```

Comment:
OK for this set of changes, fixing `peek/poll`. Please create a separate PR with only these changes, in one commit, with its unit tests. This is good to go.